switchboard/sql/SCHEMAv2.sql

```diff
@@ -0,0 +1,249 @@
+    created_at                timestamp with time zone not null,
+    expires_at                timestamp with time zone not null,
+
+    check (octet_length(token) = 128)
+);
+
+-- Privileges are assignments of permissions to subjects (users and tokens). Note that is not necessary for a token to
+-- explicitly be assigned privileges (the inheritance clause described for API tokens above).
+create table tml_switchboard.user_privileges
+(
+    user_id    uuid not null references tml_switchboard.users (user_id) on delete cascade,
+    permission text not null,
+
+    primary key (user_id, permission)
+);
+create table tml_switchboard.api_token_privileges
+(
+    token_id   uuid not null references tml_switchboard.api_tokens (token_id) on delete cascade,
+    permission text not null,
+
+    primary key (token_id, permission)
+);
+
+-- All supervisors must be registered with the switchboard database: any supervisor not registered will be turned away
+-- at the gate, so to speak.
+-- The name is at the moment superfluous and nowhere referenced, though that may change in the future.
+-- The auth_token field is quite the opposite-a supervisor attempting to connect to the switchboard must provide proof
+-- of its identity, and that is precisely what the auth_token provides: a 256-byte random string uniquely authenticating
+-- the supervisor.
+-- TODO: The tags remain ill-specified.
+create table tml_switchboard.supervisors
+(
+    supervisor_id uuid   not null primary key,
+    name          text   not null,
+    auth_token    bytea  not null unique,
+    tags          text[] not null,
+
+    check (octet_length(auth_token) = 128)
+);
+
+-- The job table is the beating heart of the switchboard.
+-- The one in the database exists for three purposes:
+--  (1) as an archive for jobs that are no longer active and have passed from the memory of the ephemeral job table in
+--      the switchboard
+--  (2) as a persistent store which can be used to reconstruct the switchboard's state in the event of a restart
+--  (3) as a compendium of the results of jobs that have finished
+--
+-- The jobs table contains all jobs that have been enqueued. Its purpose is to provide bare-bones information that can
+-- be used to restore the switchboard's functional state, and record the final exit status of all finalized jobs. It
+-- does not aim to record the concrete state of the job, beyond the minimum that is required for state reconstruction.
+
+-- Restart policy dictates the number of times a job can be restarted automatically by the switchboard.
+create type tml_switchboard.restart_policy as
+(
+    remaining_restart_count integer
+);
+-- Functional state is an internal property of jobs in the switchboard that describes how the job is treated by the
+-- switchboard; it can be treated as a simplification of execution status that only concerns itself with the
+-- requirements of the switchboard.
+create type tml_switchboard.functional_state as enum (
```

Comment:
I'm not sure whether this is a good idea. Why throw away this information just because the switchboard's scheduler currently doesn't concern itself with it? It seems simple enough to provide a reduction operator that takes the full set of job states and reduces them to these "functional state"s here.

This seems good enough though, and can always be addressed in a future migration.